src/App.test.tsx

```diff
@@ -0,0 +1,140 @@
+        expect(screen.getAllByRole('button', {name: "Delete"}).length).toBe(2)
+      });
+    });
+
+    describe('given the user clicks the create new recipe button', () => {
+      it('should redirect to the add recipe page', async () => {
+        // @ts-ignore
+        await act(async () => {
+          render(
+            <BrowserRouter>
+              <Routes>
+                <Route path="/" element='Root'></Route>
+                <Route path="/add" element='Redirected to add recipe page'></Route>
+                <Route path="/edit/1" element='Edit recipe page'></Route>
+              </Routes>
+              <App/>
+            </BrowserRouter>);
+        });
+        await act(async () => {
+          fireEvent.click(screen.getByRole('button', {name: 'Create new recipe'}));
+        });
+        expect(screen.getByText('Redirected to add recipe page')).toBeInTheDocument();
+      });
+    });
+
+    describe('given the user clicks the edit recipe button', () => {
+      it('should redirect to the add recipe page', async () => {
+        // @ts-ignore
+        await act(async () => {
+          render(
+            <BrowserRouter>
+              <Routes>
+                <Route path="/" element='Root'></Route>
+                <Route path="/edit/1" element='Redirected to edit recipe page'></Route>
+                <Route path="/add" element='Add recipe page'></Route>
+              </Routes>
+              <App/>
+            </BrowserRouter>);
+        });
+        await waitFor(() => {
+          expect(screen.getAllByRole('button', {name: "Edit"}).length).toBe(2)
+        });
+        await act(async () => {
+          const firstEditButton = screen.getAllByRole('button', {name: "Edit"})[0]
+          fireEvent.click(firstEditButton);
+        });
+        expect(screen.getByText('Redirected to edit recipe page')).toBeInTheDocument();
+      });
+    });
+
+    describe('given the user enters a search attribute and clicks the search button', () => {
+      it('should filter the results', async () => {
+        // @ts-ignore
+        await act(async () => {
+          render(
+            <BrowserRouter>
+              <Routes>
+                <Route path="/" element='Root'></Route>
+                <Route path="/recipes" element='Redirected to recipes filtered by name'></Route>
+                <Route path="/add" element='Add recipe page'></Route>
+                <Route path="/edit/1" element='Edit recipe page'></Route>
+              </Routes>
+              <App/>
+            </BrowserRouter>);
+        });
+        await act(async () => {
+          await userEvent.type(screen.getByLabelText('Name', {selector: 'input'}), 'Pizz', {delay: 0.01});
+        });
+
+        expect((screen.getByLabelText('Name', {selector: 'input'}) as HTMLInputElement).value).toBe('Pizz');
+
+        await act(async () => {
+          fireEvent.click(screen.getByRole('button', {name: "Search"}));
+        });
+
```

Comment:
One thing I miss here is an assertion to check the proper endpoint is called passing the correct params. 
For that reason in TK instead of using `msw` we mock the API methods and check they're called properly

Reply:
I completely agree :) I used `msw` because I saw it in one of the examples of the testing library but I wasn't sure how to check the params for the calls. Glad to know that in TK the API methods are mocked so that is easier to check :)